### ISSUE TITLE
Test benefits of OO interface

### ISSUE THREAD

**Issue by jmarcosfer** [MTG/essentia.js#65]
(Related to #64)

## Description
Before committing to an Object-Oriented interface of class + compute method, try out the approach outlined on #64 for only one or two algorithms (e.g. melspectrogram), and compare to the current (functional) interface.

## How?
One of the aims stated in #64, besides matching the upstream Essentia interface, is to make the library more performant.

Thus, comparison should be made between the two interface styles in terms of execution speed and/or memory.

For execution speed, something like [benchmark.js](https://benchmarkjs.com/) could be used.
For memory usage, perhaps try Chrome's `performance.measureUserAgentSpecificMemory()` or the memory profiling options on devtools.

**Comment by jmarcosfer** [comment]
@albincorreya I have now done this [here](https://github.com/MTG/essentia.js/tree/oo-interface-test/test/oo-benchmark). It's on its own branch, but it is code that I don't think will ever be merged into `dev` or `master`.

That's why I'm not opening a PR and doing the review here.

**Comment by albincorreya** [comment]
@jmarcosfer Nice. From your results, performance boost for MFCC computation is significant as expected. Maybe we could also try memory profiling on browsers just to get an idea of it?

**Comment by jmarcosfer** [comment]
Yep, definitely! I'm already working on the class-based implementation for the whole library, to make progress on that in parallel. But I would like to do memory profiling on Chrome very soon.
